MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/HomePage1/HomePagePresenter.swift

```diff
@@ -16,11 +16,11 @@ final class HomePagePresenter: HomePagePresenterProtocol {
     }
     
     func getPopularMovies() {
-        useCase.getPopularMovies { result in
+        useCase.getPopularMovies { [weak self] result in
             switch result {
             case .success(let movies):
-                let moviesViewModel: [MovieViewModel]? = self.mapMovies(movies)
-                self.delegate?.reloadCollectionView(with: moviesViewModel)
+                let moviesViewModel: [MovieViewModel]? = self?.mapMovies(movies)
+                self?.delegate?.reloadCollectionView(with: moviesViewModel)
```

Comment:
When using optional `self` more than once, you should guard it because on very very rare occasions it can happen that the self disappears between the calls.

Reply:
Guarded `self` [here](https://github.com/lihter/Movie-App/pull/24/files#diff-ab1a79673d5a383f00672fa6257f311864c75a407f50d013c3e327d71021582eR20).